batch-changes/upgrade-log4j-gradle.yml

```diff
@@ -18,7 +18,7 @@ steps:
          for proj in $PROJECTS
          do
             # Find vulnerable log4j dependency versions among all transitive dependencies.
-            VULN_LOG4J_DEPS=$(./gradlew -q "$proj":dependencies | sed -n 's/.*--- \([^ ]*\).*/\1/p' | sort | uniq | grep -E '^org.apache.logging.log4j:.*:2.((0|1|2|3|4|5|6|7|8|9|10|11|12|13|14)(\.[0-9]+)|(14\.1))$')
+            VULN_LOG4J_DEPS=$(./gradlew -q "$proj":dependencies | sed -n 's/.*--- \([^ ]*\).*/\1/p' | sort | uniq | grep -E '^org.apache.logging.log4j:.*:2.((0|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15)(\.[0-9]+)|(14\.1))$')
```

Comment:
I don't understand why we have the `(14\.1)` group in this regex because isn't it already captured by the `14|15)(\.[0-9]+)` side of the `|` or operator? I assume I'm wrong and we need this – no regex expert – but calling it out because I'm not sure if we otherwise need additional changes in light of the v15 add as well.

Reply:
Yeah I think you can drop the whole `|(14\.1)` part -- you're right, the left side should capture 2.14.##### by itself. We should probably also escape the first `.` on that left side though, since we want to match the character "." literally:

 ```suggestion
             VULN_LOG4J_DEPS=$(./gradlew -q "$proj":dependencies | sed -n 's/.*--- \([^ ]*\).*/\1/p' | sort | uniq | grep -E '^org.apache.logging.log4j:.*:2\.(0|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15)(\.[0-9]+)$')
```